SC_redesign/Crop_and_Soil/soil/nitrogen_cycling/leaching_runoff_erosion.py

```diff
@@ -0,0 +1,173 @@
+from typing import Optional
+from math import exp, log
+from SC_redesign.Crop_and_Soil.soil.soil_data import SoilData
+
+
+class LeachingRunoffErosion:
+
+    def __init__(self, soil_data: Optional[SoilData], field_size: Optional[float] = None):
+        """This method initializes the SoilData object that this module will work with, or create one if none provided.
+
+        Parameters
+        ----------
+        soil_data : SoilData, optional
+            The SoilData object used by this module to track carbon in the soil profile, creates new one if one is not
+            provided.
+        field_size : float, optional
+            Used to initialize a SoilData object for this module to work with, if a pre-configured SoilData object is
+            not provided (ha)
+
+        """
+        self.data = soil_data or SoilData(field_size=field_size)
+
+    @staticmethod
+    def _determine_nitrogen_concentration(soluble_nitrogen_amount: float,
+                                          soil_water_runoff_sum: float,
+                                          saturation_content: float) -> float:
+        """This method determines the concentration of the inorganic pools NO3/NH4 in the top soil layer
+
+        Parameters
+        ----------
+        soluble_nitrogen_amount: float
+            amount of soluble nitrogen (kg /mm H20)
+        soil_water_runoff_sum: float
+            Sum of runoff and soil water for layer (mm H2O)
+        saturation_content: float
+            volume of water in layer when saturated (mm)
+
+        Returns
+        -------
+        float
+            the concentration of the inorganic pools NO3/NH4 in the top soil layer (kg /mm H20)
+
+        References
+        -------
+        pseudocode_soil S.4.C.1
```

Comment:
```suggestion
        SWAT Theoretical documentation eqn. 4:2.1.2
        
        Notes
        -----
        This equation has been modified for use in RuFaS, it no longer accounts for the fraction of porosity from which anions are excluded.
```
We don't want any references to pseudocode_soil, unless there are absolutely no other sources for where equations or processes come from.

Also, we should add this question to a list for Pete: "Should ammonium be leached and runoff the field in the same way as nitrates?" and reference this equation. pseudocode_soil says that this equation should be used for both nitrates and ammonium, but SWAT only specifies this for use with nitrates.